model/src/main/scala/com/lightbend/scala/modelServer/model/ModelToServeStats.scala

```diff
@@ -1,47 +1,46 @@
 package com.lightbend.scala.modelServer.model
 
-import java.io.{DataInputStream, DataOutputStream}
+import java.io.DataInputStream
+import java.io.DataOutputStream
 
-import com.lightbend.model.modeldescriptor.ModelDescriptor
-
-import scala.collection.Map
-import scala.util.Try
+import scala.util.control.NonFatal
 
 /**
  * Created by boris on 5/8/17.
  */
-case class ModelToServeStats(
-  name: String = "",
-  description: String = "",
-  since: Long = 0,
-  var usage: Long = 0,
-  var duration: Double = .0,
-  var min: Long = Long.MaxValue,
-  var max: Long = Long.MinValue) {
-
-  def this(m: ModelToServe) = this(m.name, m.description, System.currentTimeMillis())
+final case class ModelToServeStats(
+  name: String,
+  description: String,
+  since: Long,
+  usage: Long = 0,
+  duration: Double = 0.0,
+  min: Long = 0,
+  max: Long = 0) {
 
   def incrementUsage(execution: Long): ModelToServeStats = {
-    usage = usage + 1
-    duration = duration + execution
-    if (execution < min) min = execution
-    if (execution > max) max = execution
-    this
+    copy(
+      usage = usage + 1,
+      duration = duration + execution,
+      min = if (execution < min) execution else min,
+      max = if (execution > max) execution else max)
```

Comment:
made it immutable

Reply:
I thought about, but would it create more load for garbage collector?

Reply:
Yes, it will be one new allocation for each increment. An example should use idiomatic and correct code. Using mutable "messages" is not.

If one would like to optimize the counters should be plain fields in the actor/stage and then the immutable ModelToServeStats copy should be created when asked for, e.g. when `sender() ! ModelToServeStats(...)`. I wouldn't do such optimizations in an educational example.